cdap-spark-core/src/main/java/co/cask/cdap/app/runtime/spark/SparkProgramRuntimeProvider.java

```diff
@@ -68,9 +69,11 @@ public ProgramRunner createProgramRunner(ProgramType type, Mode mode, Injector i
    */
   private ProgramRunner createSparkProgramRunner(Injector injector, String programRunnerClassName) {
     try {
-      CConfiguration cConf = injector.getInstance(CConfiguration.class);
-      final SparkRunnerClassLoader classLoader = new SparkRunnerClassLoader(getClassLoaderURLs(cConf),
-                                                                            getClass().getClassLoader());
+      boolean rewriteYarnClient = injector.getInstance(CConfiguration.class)
```

Comment:
It would be a bit difficult to describe since it is a combination of three bugs/behaviors that require the rewrite. I would say leave it on as long as it is working.